### PR TITLE
Adding RegNetY to TFHub

### PR DESCRIPTION
**Pull request by AdityaKane2001**
Any pull request you open is subject to the TensorFlow Hub Terms of Service at www.tfhub.dev/terms and Google's Privacy Policy at https://www.google.com/policies/privacy.

Adding RegNetY-{200, 400, 600, 800}MF to TFHub. 

Directory structure as follows:
```
assets/docs/adityakane2001
│   adityakane2001.md
│
├───collections
│   └───regnety
│           1.md
└───models
    ├───regnety200mf_classification
    │       1.md
    ├───regnety200mf_feature_extractor
    │       1.md
    ├───regnety400mf_classification
    │       1.md
    ├───regnety400mf_feature_extractor
    │       1.md
    ├───regnety600mf_classification
    │       1.md
    ├───regnety600mf_feature_extractor
    │       1.md
    ├───regnety800mf_classification
    │       1.md
    └───regnety800mf_feature_extractor
            1.md
```

The links `[3] [Colab tutorial]()` and `<!-- colab :  -->`  throughout the readme files will be filled in shortly. 

/cc @sayakpaul  @MorganR

### CODE REVIEW

**Comment by sayakpaul** [comment]
@AdityaKane2001 you need to comment on the PR to the first build error out.

File: assets/docs/adityakane2001/collections/regnety/1.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+Collection of RegNetY models trained on ImageNet-1k
+
+<!-- dataset: imagenet-ilsvrc-2012-cls -->
+<!-- task: image-classification -->
+
+## Overview
+
+This collection contains RegNetY<sup>[1]</sup> classifiers and feature extractors trained on ImageNet-1k<sup>[2]</sup>. They can be used for out-of-the box inference as well as fine-tuning. A detailed tutorial is available at <sup>[3]</sup>. Codebase used for training these models is available here<sup>[4]</sup>.
+
```
Comment (sayakpaul) [inline assets/docs/adityakane2001/collections/regnety/1.md:11]: > A detailed tutorial is available at <sup>[3]</sup>.

How about 

> A detailed tutorial is available as a Colab Notebook at <sup>[3]</sup>.

File: assets/docs/adityakane2001/models/regnety200mf_classification/1.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+model = tf.keras.Sequential([
+    hub.KerasLayer("https://tfhub.dev/adityakane2001/regnety200mf_classification/1")
+])
+predictions = model.predict(images) 
+
+# `images` must be a Tensor of the shape 
+# (batch_size, 224, 224, 3)
+# or must be generator/ iterator/ tf.data.Dataset 
+# furnishing Tensors of given dimensions. 
+```
```
Comment (sayakpaul) [inline assets/docs/adityakane2001/models/regnety200mf_classification/1.md:32]: We might want to also note that the pixel range of the input images should be [0, 255]. If you also want to add a note on running on CPU (with Group Convs) I am okay with that.

**Review by sayakpaul — CHANGES_REQUESTED** [review]

File: assets/docs/adityakane2001/models/regnety200mf_classification/1.md (hunk 2 above)
Comment (AdityaKane2001) [reply assets/docs/adityakane2001/models/regnety200mf_classification/1.md]: > ```
# `images` must be a Tensor of the shape 
# (batch_size, 224, 224, 3)
```

Actually this is incorrect. Users can pass images of any size, as we resize the images inside the model. I've made that change in the next commit. I've also added the grouped conv and [0,255] part.

File: assets/docs/adityakane2001/collections/regnety/1.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Collection adityakane2001/regnety/1
+
+Collection of RegNetY models trained on ImageNet-1k
+
+<!-- dataset: imagenet-ilsvrc-2012-cls -->
+<!-- task: image-classification -->
+
+## Overview
+
+This collection contains RegNetY<sup>[1]</sup> classifiers and feature extractors trained on ImageNet-1k<sup>[2]</sup>. They can be used for out-of-the box inference as well as fine-tuning. A detailed tutorial is available as a Colab Notebook at <sup>[3]</sup>. Codebase used for training these models is available here<sup>[4]</sup>.
```
[the remark this answers is not in the document]
Comment (AdityaKane2001) [reply assets/docs/adityakane2001/collections/regnety/1.md:10]: I feel this is consistent with some of the other files in the repo. But please tell me if you think I should use links instead.

File: assets/docs/adityakane2001/collections/regnety/1.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+
+## Table of contents
+| Sr no | Model Name    | Accuracy | Inference speed on K80 | Inference speed on V100 | FLOPs (Number of parameters) | Link                                                                                                                                                                 |
```
[the remark this answers is not in the document]
Comment (AdityaKane2001) [reply assets/docs/adityakane2001/collections/regnety/1.md:14]: I'll remove it.

File: assets/docs/adityakane2001/models/regnety200mf_classification/1.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+RegNetY 200MF image classifier pretrained on ImageNet-1k.
+
+<!-- asset-path: https://storage.googleapis.com/ak-regnety-savedmodels/tars/regnety200mf_classification.tar.gz  -->
+<!-- task: image-classification -->
+<!-- network-architecture: regnety -->
+<!-- format: saved_model_2 -->
+<!-- fine-tunable: false -->
+<!-- license: apache-2.0 -->
+<!-- language: en -->
+<!-- colab:    -->
```
[the remark this answers is not in the document]
Comment (AdityaKane2001) [reply assets/docs/adityakane2001/models/regnety200mf_classification/1.md:12]: The notebook is ready to go. Actually I'm just waiting for your approval on the same 😅. ([link to PR](https://github.com/AdityaKane2001/regnety/pull/18))

File: assets/docs/adityakane2001/models/regnety200mf_classification/1.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+```python
+model = tf.keras.Sequential([
+    hub.KerasLayer("https://tfhub.dev/adityakane2001/regnety200mf_classification/1")
+])
+predictions = model.predict(images) 
+```
+
+Inputs to the model must:
+1. be four dimensional Tensors of the shape `(batch_size, height, width, num_channels)`. Note that the model expects images with  `channels_last`  property. `height`,`width` and `batch_size` can be any value based on user's preference.
+2. have pixel values in the range `[0, 255]`.
```
[the remark this answers is not in the document]
Comment (AdityaKane2001) [reply assets/docs/adityakane2001/models/regnety200mf_classification/1.md:31]: We later moved normalization part to inside the model because of performance issues. IMO this provides with an added advantage, for direct classification one doesn't have to resize the images to (224, 224). But please tell me if you think it's better to remove that part. For reference, please see [PreStem](https://github.com/AdityaKane2001/regnety/blob/4e41a3c56a623d6c6fca1dff0e425c5143cbfe6b/regnety/models/blocks.py#L23-L55) class.

File: assets/docs/adityakane2001/models/regnety200mf_feature_extractor/1.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
+# Module adityakane2001/regnety200mf_feature_extractor/1
+
+RegNetY 200MF feature extractor pretrained on ImageNet-1k.
+
+<!-- asset-path:  https://storage.googleapis.com/ak-regnety-savedmodels/tars/regnety200mf_feature_extractor.tar.gz -->
+<!-- task: image-classification -->
+<!-- network-architecture: regnety -->
+<!-- format: saved_model_2 -->
+<!-- fine-tunable: false -->
```
[the remark this answers is not in the document]
Comment (AdityaKane2001) [reply assets/docs/adityakane2001/models/regnety200mf_feature_extractor/1.md:9]: Sorry, it's a mistake. I'll change that.

File: assets/docs/adityakane2001/models/regnety200mf_classification/1.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (MorganR) [reply assets/docs/adityakane2001/models/regnety200mf_classification/1.md:31]: Ah that's a shame, no worries. Agreed that accepting dynamic image sizes is simpler for the user. It might still be worth mentioning that they are resized to 224x224 internally so that users don't use unnecessarily large images.

File: assets/docs/adityakane2001/models/regnety200mf_classification/1.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+Model outputs can be inferred based on mapping of integers to labels as in here<sup>[2]</sup>. 
+
+For downstream task that involves fine-tuning, it is recommended to use a feature extractor of this model. Fine tuning this model is explained in accompanying Colab notebook.
+
+One known caveat of this implementation is as of TensorFlow 2.5, this model can run only on GPU or TPU and not CPU. This is because grouped convolutions used in this model are not supported on CPUs be TensorFlow. However, from TensorFlow 2.6.0rc1 grouped convs with batch_size=1  are supported on CPUs.
```
[the remark this answers is not in the document]
Comment (MorganR) [reply assets/docs/adityakane2001/models/regnety200mf_classification/1.md:37]: Please update this section accordingly.

**Review by MorganR — CHANGES_REQUESTED** [review]

File: assets/docs/adityakane2001/models/regnety200mf_classification/1.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (MorganR) [reply assets/docs/adityakane2001/models/regnety200mf_classification/1.md:40]: I think it would help to add a note and link to the keras issue. As written, users would expect TF 2.6 to work, so they may be confused.

**Review by MorganR — CHANGES_REQUESTED** [review]
One more change requested to link to the keras issue.

File: assets/docs/adityakane2001/models/regnety200mf_classification/1.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (AdityaKane2001) [reply assets/docs/adityakane2001/models/regnety200mf_classification/1.md]: @MorganR 

Added comment about this.

```
At the time of writing this, grouped convs are not functional at all in Keras with TensorFlow 2.6.0. You can track the progress on this [here](https://github.com/keras-team/keras/issues/15162).
```

**Review by MorganR — CHANGES_REQUESTED** [review]
Please add the note about the keras issue to the other models as well.

**Comment by AdityaKane2001** [comment]
@MorganR 

I tested all models once again. They are good to go. Please let me know if there is any problem.

**Comment by AdityaKane2001** [comment]
@MorganR 

Changed `adityakane2001` to `AdityaKane2001` wherever relevant. Could you please run the workflow again?

File: assets/docs/adityakane2001/collections/regnety/1.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
+# Collection adityakane2001/regnety/1
```
Comment (MorganR) [inline assets/docs/adityakane2001/collections/regnety/1.md:1]: Need to fix capitalization of adityakane here

File: assets/docs/adityakane2001/adityakane2001.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
+# Publisher AdityaKane2001
```
Comment (MorganR) [inline assets/docs/adityakane2001/adityakane2001.md:1]: If you want "AdityaKane2001" to have capital letters, you need to change the directory name across all these markdown files to match (and in this case, the .md file name too)

**Review by MorganR — CHANGES_REQUESTED** [review]

**Comment by MorganR** [comment]
Btw, you can pre-check that your markdown passes validation. You can execute the same checks locally by passing the file paths of modified Markdown files (relative to the `assets/docs` directory) to validator.py e.g. `python3 ./tools/validator.py google/google.md google/models/albert_base/1.md ...`.

File: assets/docs/adityakane2001/adityakane2001.md (hunk 10 above)
Comment (MorganR) [reply assets/docs/adityakane2001/adityakane2001.md:1]: Actually, sorry for this, by convention we prefer lowercase publisher names. Could you please swap everything back to lowercase?

Comment (AdityaKane2001) [reply assets/docs/adityakane2001/adityakane2001.md:1]: @MorganR 

I want to keep `adityakane2001` everywhere, following the repo convention. In that case, I should keep all filenames and contents inside them as `adityakane2001` , right?

**Comment by AdityaKane2001** [comment]
@MorganR 

I have made the changes. The errors were caused by file endings. As I pushed the code from Windows, the endings were CRLF, which are not acceptable. Changed them to LF and changed username to lowercase.

**Comment by sayakpaul** [comment]
@MorganR possible to merge this and run the workflow?

File: assets/docs/adityakane2001/adityakane2001.md (hunk 10 above)
Comment (MorganR) [reply assets/docs/adityakane2001/adityakane2001.md]: Correct, this looks correct now.

**Review by MorganR — APPROVED** [review]

**Review by MorganR — CHANGES_REQUESTED** [review]
Aditya, can you delete these pycache files? 

tools/__pycache__/filesystem_utils.cpython-39.pyc
tools/__pycache__/yaml_parser.cpython-39.pyc

**Comment by AdityaKane2001** [comment]
@MorganR 

I've removed the pycache and squashed the commits.

**Review by MorganR — APPROVED** [review]

**Comment by AdityaKane2001** [comment]
@MorganR 

Could you please merge this one?

**Comment by AdityaKane2001** [comment]
Thanks a lot!
